File: 백준/Bronze/3052. 나머지/나머지.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] arr = new int[42];
        int n, count = 0;
        for(int i = 0; i < 10; i++) {
            n = sc.nextInt();
            arr[n % 42]++;
        }
        for(int i = 0; i < 42; i++)
            if(arr[i] != 0) count++;
        System.out.println(count);
    }
}

File: 백준/Bronze/3052. 나머지/나머지.cc
#include <stdio.h>

int main(void) {
	int arr[42] = { 0 }, n, count = 0;
	for (int i = 0; i < 10; i++) {
		scanf("%d", &n);
		arr[n % 42]++;
	}
	for (int i = 0; i < 42; i++)
		if (arr[i] != 0) count++;
	printf("%d", count);
	return 0;
}